keras/layers/recurrent.py

```diff
@@ -217,7 +218,15 @@ def call(self, x, mask=None):
                                 'argument, including the time axis. '
                                 'Found input shape at layer ' + self.name +
                                 ': ' + str(input_shape))
-        if self.stateful:
+        if self.initial_states is not None:
+            if self.stateful:
+                raise Exception('Keras does not currently support stateful Recurrent '
+                                'layers with custom initial states.')
+            if type(self.initial_states) in (list, tuple):
+                initial_states = self.initial_states
+            else:
+                initial_states = [self.initial_states]
```

Comment:
Right. The use-case is [sequence to sequence models](https://papers.nips.cc/paper/5346-sequence-to-sequence-learning-with-neural-networks.pdf). In those models, the final hidden state of the first set of layers is the initial hidden state for the subsequent set of layers. There are some other features that are needed to fully implement the model described in that paper, but I'm currently working on them, so I should have another pull request ready soon.